listlayout.py

```diff
@@ -1,24 +1,53 @@
 from layout import Layout
 from pos import Pos
+from size import Size
 
 import pygame
 import colour
 
 class ListLayout(Layout):
-    def __init__(self, size, pos=Pos(0,0)):
-        super(ListLayout, self).__init__(size, pos)
+    def __init__(self, size, pos=Pos(0,0), border=5, orientation="vertical", spacing=2):
+        super(ListLayout, self).__init__(size, pos, border)
         self.next_child_origin = Pos(
-            self.origin.x + self.pos.x, 
-            self.origin.y + self.pos.y
+            self.origin.x + self.pos.x + self.border, 
+            self.origin.y + self.pos.y + self.border
         )
+        self.orientation = orientation
+        self.spacing = spacing
 
     def add(self, child):
+
+        if not self.children:
+            self.size = Size(
+                child.size.w + 2*self.border,
+                child.size.h + 2*self.border
+            )
+        else:
+            if self.orientation == "horizontal":
```

Comment:
no need for two giant blocks here. We can just set the size once but populate the values conditionally